src/Sylius/Behat/Context/Api/Shop/ProductVariantContext.php

```diff
@@ -274,6 +302,50 @@ public function variantAndVariantShouldBeVisible(array $variants): void
         }
     }
 
+    /**
+     * @Then /^I should see variant with ("[^"]+" option) and ("[^"]+" option value) priced at ("[^"]+") at (\d)(?:st|nd|rd|th) position$/
+     */
+    public function iShouldSeeVariantWithOptionPricedAtAtPosition(
+        string $expectedOptionName,
+        string $expectedOptionValueValue,
+        int $price,
+        int $position,
+    ): void {
+        $variants = $this->sharedStorage->get('product_variant_collection');
+        Assert::notNull($variants);
+        Assert::greaterThan(count($variants), $position - 1, 'There are less variants than expected');
+
+        $variant = $variants[$position - 1];
+        Assert::same($variant['price'], $price);
+
+        $optionValueData = $this->fetchItemByIri($variant['optionValues'][self::FIRST_COLLECTION_ITEM]);
```

Comment:
It should be iterated over.
Given the step definition, it should work when we check against a variant with more than one option value, and the one we specify is not the first.